deployment/ccip/changeset/v1_5_1/cs_deploy_ping_pong.go

```diff
@@ -0,0 +1,198 @@
+		router = chainState.TestRouter
+	}
+
+	if router == nil {
+		return fmt.Errorf("router address is empty for chain %d", c.ChainSelector)
+	}
+
+	if chainState.LinkToken == nil {
+		return fmt.Errorf("link token address is empty for chain %d", c.ChainSelector)
+	}
+
+	return nil
+}
+
+func DeployPingPongDemoContractsChangeset(env deployment.Environment, c DeployPingPongDemoContractsConfig) (deployment.ChangesetOutput, error) {
+	state, err := changeset.LoadOnchainState(env)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to load onchain state: %w", err)
+	}
+
+	if err := c.Validate(env, state); err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("invalid DeployPingPongDemoContractsConfig: %w", err)
+	}
+
+	newAB := deployment.NewMemoryAddressBook()
+
+	chain := env.Chains[c.ChainSelector]
+	chainState := state.Chains[c.ChainSelector]
+
+	router := chainState.Router
+	if c.IsTestRouter {
+		router = chainState.TestRouter
+	}
+
+	tv := deployment.NewTypeAndVersion(changeset.PingPongDemo, deployment.Version1_0_0)
+
+	addr, _, _, err := ping_pong_demo.DeployPingPongDemo(chain.DeployerKey, chain.Client, router.Address(), chainState.LinkToken.Address())
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to deploy ping pong demo contract on %d: %w", c.ChainSelector, err)
+	}
+
+	err = newAB.Save(chain.Selector, addr.String(), tv)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to save ping pong demo contract address: %w", err)
+	}
+
+	return deployment.ChangesetOutput{
+		AddressBook: newAB,
+	}, nil
+}
+
+type StartPingPongDemoContractsConfig struct {
+	ChainSelector uint64
+}
+
+func (c StartPingPongDemoContractsConfig) Validate(env deployment.Environment, state changeset.CCIPOnChainState) error {
+	chainState := state.Chains[c.ChainSelector]
+
+	if chainState.PingPongDemo == nil {
+		return fmt.Errorf("ping pong demo address is empty for chain %d", c.ChainSelector)
+	}
+
+	return nil
+}
+
+func StartPingPongDemoContractsChangeset(env deployment.Environment, c StartPingPongDemoContractsConfig) (deployment.ChangesetOutput, error) {
+	state, err := changeset.LoadOnchainState(env)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to load onchain state: %w", err)
+	}
+
+	if err := c.Validate(env, state); err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("invalid StartPingPongDemoContractsConfig: %w", err)
+	}
+
+	chain := env.Chains[c.ChainSelector]
+	chainState := state.Chains[c.ChainSelector]
+
+	transactor, err := ping_pong_demo.NewPingPongDemoTransactor(chainState.PingPongDemo.Address(), chain.Client)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to create transactor for ping pong demo: %w", err)
+	}
+
+	_, err = transactor.StartPingPong(chain.DeployerKey)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to start ping pong demo: %w", err)
+	}
```

Comment:
All txs need to be verified with [ConfirmIfNoErrorWithABI](https://github.com/smartcontractkit/chainlink/blob/develop/deployment/environment.go#L301)